nncf/pruning/filter_pruning/algo.py

```diff
@@ -280,62 +282,82 @@ def _find_layerwise_pruning_rate(self, target_flops_pruning_rate):
         if flops < target_flops:
             self.current_flops = flops
             return right
-        raise RuntimeError("Can't prune model to asked flops pruning rate = {}".format(target_flops_pruning_rate))
+        raise RuntimeError("Can't prune the model to get the required "
+                           "pruning rate in flops = {}".format(target_flops_pruning_rate))
 
-    def set_pruning_rate(self, pruning_rate, run_batchnorm_adaptation=False):
+    def set_pruning_rate(self, pruning_rate: Union[float, Dict],
+                         run_batchnorm_adaptation: bool = False, groupwise_pruning_rates: bool = False):
```

Comment:
Possible defined `groupwise_pruning_rates` by type of `pruning_rate`, instead of additional argument

Reply:
And i dont find calling set_pruning_rate with `groupwise_pruning_rates`

Reply:
done. we not not use `set_pruning_rate` with a dict internally right now. in the future possibly in pair with hyperparameter search algos.